### ISSUE TITLE
Basic auth will be deprecated

### ISSUE THREAD

**Issue by atmartins** [singer-io/tap-jira#43]
Per https://developer.atlassian.com/cloud/jira/platform/deprecation-notice-basic-auth-and-cookie-based-auth/, basic auth will no longer be supported. Suggestion is to replace the basic auth feature here with api token authorization. The readme should be updated as well, with example config.

**Comment by luandy64** [comment]
I agree with you on the README update, but only to say that for most people to use OAuth tokens.

We need to keep support for basic auth for the people that run this tap against an on-prem Jira instance. Unless I'm wrong about this.

**Comment by atmartins** [comment]
Ah okay, that makes sense. I think the Oauth stuff is pretty difficult to set up, as Atlassian doesn't seem to provide all those values in one place, or with those names. Do you think it would be beneficial to add api token authorization in addition to Basic and OAuth? I'm happy to submit a PR soon if you think that is a good idea.

**Comment by luandy64** [comment]
Atlassian provides a [tutorial](https://developer.atlassian.com/cloud/jira/service-desk/jira-rest-api-oauth-authentication/) on getting the OAuth credentials. 

So I think if you go through the tutorial and get [these values](https://github.com/singer-io/tap-jira/blob/e25f4e2214239a899d0809e2ca30cc96d87b356d/tap_jira/__init__.py#L14-L19), then putting those values in a `my_tap_config.json` should get the tap to run.

If you think the api token auth PR would be easier than the tutorial I linked, feel free to open it.

**Comment by rpaterson** [comment]
Just tried it and you can specify the auth token instead of `password` and it works fine. So no code changes needed, maybe just update the README.

**Comment by KBorders01** [comment]
@diegojardon did you select any streams in your catalog JSON file? Putting the API token in the password field with basic authentication works for me as of now.
